growattServer/__init__.py

```diff
@@ -580,33 +583,63 @@ def update_plant_settings(self, plant_id, changed_settings, current_settings = N
         data = json.loads(response.content.decode('utf-8'))
         return data
 
-    def update_mix_inverter_setting(self, serial_number, setting_type, parameters):
+
+    def update_inverter_setting(self, serial_number, setting_type, 
+                                default_parameters, parameters, endpoint):
         """
         Applies settings for specified system based on serial number
         See README for known working settings
 
         Keyword arguments:
-        serial_number -- The serial number (device_sn) of the inverter
-        setting_type -- The setting to be configured (list of known working settings below)
-        parameters -- A python dictionary of the parameters to be sent to the system based on the chosen setting_type, OR a python array which will be converted to a params dictionary
+            serial_number (str): Serial number (device_sn) of the inverter
```

Comment:
Yes, that's a good point. Been a while since I looked at [PEP257](https://peps.python.org/pep-0257/#multi-line-docstrings). There doesn't seem to be a convention for colons versus double hyphens, but you're right about the indentations. I've now written them to be consistent with the other functions. At some point, I'll draft a docstring for the module. I first want to sort out how to get the data output as described in #33.

Reply:
Thanks, my OCD is now satisfied :)

It's all good from my side now. I know it's clunky but I do recommend using the netssl app on android to work out what endpoint the app hits.

P.S. As part of the home-assistant integration myself and Indy maintain I'm also looking at adding support for the types we don't currently support as well, going to collate them all next week, so there may be some overlap - I'll keep you posted.